src/AllocationIDTracker.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+/**
+ * @title AllocationIDTracker
+ * @dev This contract tracks the allocation IDs of the RAVs that have been submitted to
+ *      ensure that each allocation ID is only used once. It is external to collateral
+ *      contract to allow for updating the collateral contract without losing the list of
+ *      used allocation IDs.
+ * @notice This contract is intended to be used with the `Collateral` contract.
+ */
+contract AllocationIDTracker {
+    mapping (address => bool) private _usedAllocationIDs;
```

Comment:
Current `AllocationIDTracker` assumes a single gateway. We can future proof it by modifying the `_usedAllocationIDs` mapping to make the pair (allocation id, gateway) unique instead of just the allocation id. So maybe we can rename that contract to `RAVTracker` and make the mapping a mapping of a mapping where the first key is the gateway (sender on the redeem function) and the second key is the allocation id.

Reply:
This will be tracked in #8